### PR TITLE
docs: add shim config option, add missing args docs

### PR DESCRIPTION
**Pull request by thepassle**
depends on https://github.com/withastro/astro/pull/3317

### CODE REVIEW

File: src/pages/en/reference/adapter-reference.md [hunk 1]
```diff
@@ -42,15 +42,19 @@ The object passed into `setAdapter` is defined as:
...
 }
 ```
 
 The properties are:
 
 * __name__: A unique name for your adapter, used for logging.
 * __serverEntrypoint__: The entrypoint for server-side rendering.
+* __shim__: Optional array of import specifiers pointing to polyfills/shims.
 * __exports__: An array of named exports when used in conjunction with `createExports` (explained below).
+* __args__: Optional args to pass to your server handler.
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:57]: If it's correct to say "arguments" in the property description, then I would suggest using the unabbreviated word here.

Comment (thepassle) [reply src/pages/en/reference/adapter-reference.md:57]: The `args` property already existed, it was just undocumented. Changing it to `arguments` now would be a breaking change.

Comment (delucis) [reply src/pages/en/reference/adapter-reference.md:57]: I think Sarah means in the explanation of the option (rather than changing the property name).

Comment (delucis) [reply src/pages/en/reference/adapter-reference.md:57]:
```suggestion
* __args__: Optional arguments to pass to your server handler.
```

**Comment by delucis** [comment]
Good thinking @sarah11918 — done! @thepassle please mark this as ready for review once https://github.com/withastro/astro/pull/3317 is merged. Thanks!

**Pull request closed without merging**